code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -8,290 +9,140 @@
 
-	spawn(5)
-		.()
-
-/obj/item/weapon/pinpointer/operative
-	name = "operative pinpointer"
-	icon = 'icons/obj/device.dmi'
-	desc = "A pinpointer that leads to the first Syndicate operative detected."
-	var/mob/living/carbon/nearest_op = null
-
-/obj/item/weapon/pinpointer/operative/attack_self()
-	if(!usr.mind || !(usr.mind in ticker.mode.syndicates))
-		usr << "<span class='danger'>AUTHENTICATION FAILURE. ACCESS DENIED.</span>"
-		return 0
-	if(!active)
-		active = 1
-		workop()
-		usr << "<span class='notice'>You activate the pinpointer.</span>"
-	else
-		active = 0
-		icon_state = "pinoff"
-		usr << "<span class='notice'>You deactivate the pinpointer.</span>"
-
-/obj/item/weapon/pinpointer/operative/proc/scan_for_ops()
-	if(active)
-		nearest_op = null //Resets nearest_op every time it scans
-		var/closest_distance = 1000
-		for(var/mob/living/carbon/M in mob_list)
-			if(M.mind && (M.mind in ticker.mode.syndicates))
-				if(get_dist(M, get_turf(src)) < closest_distance) //Actually points toward the nearest op, instead of a random one like it used to
-					nearest_op = M
-
-/obj/item/weapon/pinpointer/operative/proc/workop()
-	if(active)
-		scan_for_ops()
-		point_at(nearest_op, 0)
-		spawn(5)
-			.()
-	else
-		return 0
-
-/obj/item/weapon/pinpointer/operative/examine(mob/user)
-	..()
-	if(active)
-		if(nearest_op)
-			user << "Nearest operative detected is <i>[nearest_op.real_name].</i>"
-		else
-			user << "No operatives detected within scanning range."
+/obj/item/weapon/pinpointer/proc/my_god_jc_a_bomb() //If we should get the hell back to the ship
+	for(var/obj/machinery/nuclearbomb/bomb in machines)
+		if(bomb.timing)
+			icon_state = "pinonalert"
+			if(!nuke_warning)
+				nuke_warning = TRUE
+				playsound(src, 'sound/items/Nuke_toy_lowpower.ogg', 50, 0)
+				if(isliving(loc))
+					var/mob/living/L = loc
+					L << "<span class='userdanger'>Your [name] vibrates and lets out a tinny alarm. Uh oh.</span>"
+
+/obj/item/weapon/pinpointer/proc/switch_mode_to(new_mode) //If we shouldn't be tracking what we are
+	if(isliving(loc))
+		var/mob/living/L = loc
+		L << "<span class='userdanger'>Your [name] beeps as it reconfigures its tracking algorithms.</span>"
+		playsound(L, 'sound/machines/triple_beep.ogg', 50, 1)
+	mode = new_mode
+	target = null //Switch modes so we can find the new target
+
+/obj/item/weapon/pinpointer/proc/refresh_target() //Periodically removes the target to allow the pinpointer to update (i.e. malf AI shunts, an operative dies)
+	target = null
+
+/obj/item/weapon/pinpointer/syndicate //Syndicate pinpointers automatically point towards the infiltrator once the nuke is active.
+	name = "syndicate pinpointer"
+	desc = "A handheld tracking device that locks onto certain signals. It's configured to switch tracking modes once it detects the activation signal of a nuclear device."
+
+/obj/item/weapon/pinpointer/syndicate/cyborg //Cyborg pinpointers just look for a random operative.

```

Comment:
Would it be possible to allow the borg to cycle through the Ops using attack_self?